index.bs

```diff
@@ -156,53 +160,53 @@ spec:fetch; type:dfn; text:value
   </section>
   <section>
     <h3 id="policies">Policies</h3>
-    <p>A {{Document}} has a <dfn>feature policy</dfn>, which consists of:</p>
+    <p>A <dfn>feature policy</dfn> is a struct with the following items:</p>
     <ul>
-      <li>A set of <a data-lt="inherited policy set">inherited policies</a>.
+      <li>An <a data-lt="inherited policy">inherited policy</a>.
       </li>
       <li>A <a data-lt="declared policy">declared policy</a>.
       </li>
     </ul>
-    <p>An <dfn>empty feature policy</dfn> has an <a>inherited policy set</a>
+    <p>An <dfn>empty feature policy</dfn> has an <a>inherited policy</a>
     which contains 'Enabled' for every <a>policy-controlled feature</a>
     available, and a <a>declared policy</a> which is an empty map.</p>
   </section>
   <section>
     <h3 id="inherited-policies">Inherited policies</h3>
+    <p>An <dfn data-lt="inherited policy|inherited policies">inherited
+    policy</dfn> is an ordered map from <a
+    data-lt="policy-controlled feature">features</a> to either 'Enabled' or
+    'Disabled'.</p>
+    <p>The <dfn>inherited policy for a feature</dfn> <var>feature</var> is the
+    value in the <a>inherited policy</a> whose key is <var>feature</var>. After
+    a <a>feature policy</a> has been initialized, its <a>inherited policy</a>
+    will contain an value for each <a>supported feature</a>.</p>
+    <div class="note">
     <p>Each document in a frame tree inherits a set of policies from its parent
     frame, or in the case of the top-level document, from the defined defaults
-    for each <a>policy-controlled feature</a>. This inherited policy set
-    determines the initial state (enabled or disabled) of each feature, and
-    whether it can be controlled by a <a>declared policy</a> in the document.
+    for each <a>policy-controlled feature</a>. This inherited policy determines
+    the initial state (enabled or disabled) of each feature, and whether it can
+    be controlled by a <a>declared policy</a> in the document.
     </p>
     <p>In a {{Document}} in a [=top-level browsing context=], the inherited
-    feature set is based on defined defaults for each feature.</p>
-    <p>In a {{Document}} in a [=nested browsing context=], the inherited feature
-    set is based on the parent document's feature policy, as well as any
+    policy is based on defined defaults for each feature.</p>
+    <p>In a {{Document}} in a [=nested browsing context=], the inherited policy
+    is based on the parent document's feature policy, as well as any
     <a href="#iframe-allow-attribute">allow attributes</a> defined on the
     browsing context container.</p>
```

Comment:
This isn't entirely correct. This only applies when the container is an `iframe`, right?

Reply:
I think that the way the spec is written, this should apply to all nested browsing contexts -- it would be useful for `<frames>` to have the same restrictions (as well as `<object>` when used to contain HTML docs; no idea about SVG docs) as `<iframes>`. If an API has been determined to not be usable in cross-origin `<iframes>`, you shouldn't be able to use it in a cross origin `<frame>`.

The only difference with `iframe`s, I think, is that only `iframe`s have the extra `allow`, `allowfullscreen`, `allowusermedia`, and `allowpaymentrequest` attributes. So the "Process feature policy attributes" algorithm treats iframes specially.

The Chrome implementation uses the parent's policy to define an inherited policy for *all* nested browsing contexts. but only defines `allow` on `<iframe>`.

Reply:
I've created another patch, to reword this, and to change the term 'frame' where appropriate to 'nested browsing context', since 'frame' has a different meaning in HTML. (I suppose it has three or four different meanings in HTML, but they're all different than this)

https://github.com/clelland/feature-policy/pull/1

Reply:
Thanks, I did mean the bit about the allow attribute. I agree that restrictions should apply regardless.